### ISSUE TITLE
What is the behavior if a JSON path is an array, object or literal in different cases?

### ISSUE THREAD

**Issue by jpmckinney** [kindly/flatterer#14]
I expect an error as behavior is undefined – but better if the error is human-readable and clear to a user.

**Comment by jpmckinney** [comment]
This works as follows:

```
[
    {
        "foo": "bar"
    },
    {
        "foo": {
            "baz": "bzz"
        }
    },
    {
        "foo": [
            {
                "zzz": 1
            }
        ]
    }
]
```

yields:

`main.csv`

```csv
_link,_link_main,foo,foo_baz
1,1,bar,
2,2,,bzz
3,3,,
```

`foo.csv`

```csv
_link,_link_main,zzz
3.foo.0,3,1
```

**Comment by kindly** [comment]
More work needs to be done on documenting output format. #5 

However, the above is intended behaviour as I wanted to accept the most amount of possible input data, even with inconsistant keys.  Probably a warning would be useful though.

**Comment by kindly** [comment]
Closing not fixed will be part of #5.
